// proto_definitions\ml_stock_service.proto
syntax = "proto3";

// サービスの定義
service MLStockService {
  rpc GetMLStockData (MLStockRequest) returns (MLStockResponse);
}

// リクエストメッセージの定義
message MLStockRequest {
  repeated string symbols = 1;
  string start_date = 2;
  string end_date = 3;
}

// レスポンスメッセージの定義
message MLStockResponse {
  repeated MLSymbolData symbol_data = 1;  // シンボルごとのデータリスト
}

// シンボルごとのデータとシグナルを格納するメッセージの定義
message MLSymbolData {
  string symbol = 1;  // シンボル名
  repeated MLDailyData daily_data = 2;  // 日足データリスト
  repeated string signals = 3;  // 予測結果の日付リスト
}

// 日足データの定義
message MLDailyData {
  string date = 1;
  float open = 2;
  float high = 3;
  float low = 4;
  float close = 5;
  int64 volume = 6;
}


// protoc --python_out=. proto_definitions/ml_stock_service.proto
